File: EmailListenerService/src/main/java/services/CreateCamundaTaskService.java
package services;


import java.io.IOException;
import java.net.URI;
import java.net.URISyntaxException;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class CreateCamundaTaskService {
    private static final String _url = "localhost:8080/engine-rest/process-definition/key/handle-complaint/start";

    public static void CreateCamundaTask(String email, String content) throws IOException, URISyntaxException, InterruptedException {
        System.out.println("Ive been called.");
        String jsonInput = String.format("{\"variables\": {\"complaint\": {\"value\": \"%s\", type: \"String\"}, \"email\": {\"value\": \"%s\", type: \"String\"}}, \"businessKey\": \"javaemaillistener\"}", content, email);
        HttpRequest request = HttpRequest.newBuilder()
                .uri(new URI(_url))
                .POST(HttpRequest.BodyPublishers.ofString(jsonInput))
                .build();
        HttpClient client = HttpClient.newHttpClient();
        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
        System.out.println(response);
    }
}

File: EmailListenerServiceNET/EmailListenerService/Services/CreateCamundaTaskService.cs
using EmailListenerService.Model;
using System.Net.Http.Headers;
using Newtonsoft.Json;

namespace EmailListenerService.Services
{
    public class CreateCamundaTaskService
    {
        private const string _url = "http://localhost:8080/engine-rest/process-definition/key/handle-complaint/start";
        static readonly HttpClient client = new();
        public static void CreateCamundaTask(String email, String emlcontent) {
            Console.WriteLine("Ive been called.");
            

            try {
                //string jsonInput = "{\"variables\": {\"complaint\": {\"value\": \"" + emlcontent+"\", \"type\": \"String\"}, \"email\": {\"value\": \""+email+"\", \"type\": \"String\"}}, \"businessKey\": \".netemaillistener\"}";

                ComplaintDTO complaint = new ComplaintDTO(emlcontent, email);
                string jsonInput = JsonConvert.SerializeObject(complaint);
                Console.WriteLine(jsonInput);
                HttpContent content = new StringContent(jsonInput, System.Text.Encoding.UTF8, "application/json");
                content.Headers.ContentType = new System.Net.Http.Headers.MediaTypeHeaderValue("application/json");
                client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
                client.DefaultRequestHeaders.Connection.Add("keep-alive");
                var response = client.PostAsync(_url, content).Result;
                Console.WriteLine(response);
            }
            catch(Exception ex)
            {
                Console.WriteLine(ex.Message);
            }
            
        }
}
}
